Sources/Build/BuildPlan.swift

```diff
@@ -229,6 +268,14 @@ public final class ClangTargetBuildDescription {
         }
         args += ["-fblocks"]
 
+        // Enable index store, if appropriate.
+        //
+        // Currently, we only enable index store for Apple's clang because it is
+        // not widely available in OSS clang.
+        if buildParameters.triple.isDarwin(), (try? buildParameters.toolchain._isClangCompilerVendorApple()) == true {
+            args += buildParameters.indexStoreArguments
+        }
```

Comment:
AFAICT there's no way to enable the flag even if you *know* your clang supports it (for example, we could potentially put a clang that supports it in the linux OSS toolchain).
How about making it so that if the user passes '--enable-index-store' that it enables the flag regardless if the clang compiler is from Apple or not ?

Reply:
Hmm, maybe separate `--enable-clang-index-store` option, otherwise if someone wants to pass `--enable-index-store` only to enable it for release, it could cause clang failures.

Reply:
An env var seems fine to me, but does it need to be prefixed with `SWIFTPM_` ?

Reply:
Yep.